feat (components/button): add Button

sgrisshk opened the pull request:
## Status

**READY**

## Description

add new component 'button' in two (filled/bordered) styles from figma

## Type of Change

<!--- Put an `x` in all the boxes that apply: -->

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

sgrisshk committed:
feat (elements): add new component

add new component 'button' in two (filled/bordered) styles from figma

Robocotik commented on src/components/Button/button.tsx:
```diff
@@ -0,0 +1,21 @@
+import { ButtonProps } from './button.props';
+import '../Text/index'
+const Button: React.FC<ButtonProps> = ({ label, disabled = false, color = 'primary' }) => {
+    const baseStyles = 'px-4 py-2 font-headline-small rounded';
```
я бы не стал выносить такие стили куда-то вверх, ибо честно не верю, что там прям так сильно все меняется , я бы использовал тернарную операцию для такого,

coma8765 replied:
хранить константы не в функции а перед ней

Robocotik commented on src/components/Button/button.tsx:
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { ButtonProps } from './button.props';
```
мы props компонентов называем всегда Props. Всегда. в каждой папке компонента он такой один будет)

Robocotik commented on src/components/Button/button.tsx:
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { ButtonProps } from './button.props';
+import '../Text/index'
+const Button: React.FC<ButtonProps> = ({ label, disabled = false, color = 'primary' }) => {
```
не хватает ...props и className в пропсах, их ты должна получить из extend, которого у тебя нет

Robocotik replied:
более быстрый вариант писать const Button: FC<Props> = ({бла бла бла}) => { return()}

coma8765 replied:
`...props`

Robocotik commented on src/components/Button/button.tsx:
```diff
@@ -0,0 +1,21 @@
+
+    return (
+        <button
+            className={`${baseStyles} ${buttonStyles} ${disabled}`}
```
className надо таакже пробросить сюда из  props

coma8765 commented on src/components/Button/button.stories.tsx:
```diff
@@ -0,0 +1,22 @@
+Secondary.args = {
+    label: 'Мои Боты',
+    color: 'bordered',
+};
```
интересный способ представление сторис, но у нас другой формат

респект за собственное изучение либы
но нужно переделать как делается в других местах, это единый стиль

coma8765 replied:
@Robocotik добавь в доку плиз что мы понимаем, что можно написать одну и туже вещь разными способами, но крайне не рекомендуем так делать, чтоб был единый стиль в коде

Robocotik commented on src/components/Button/button.props.ts:
```diff
@@ -0,0 +1,5 @@
+export interface ButtonProps {
```
не хватает exnteds HTMLAttribute<HTMLButtonElement>

coma8765 replied:
расширение от div или от кнопки

а то как прикидывать className и onClick

coma8765 commented on src/components/Button/button.stories.tsx:
```diff
@@ -0,0 +1,22 @@
+    component: Button,
+} as Meta<typeof Button>;
+
+const Template: StoryFn<typeof Button> = (args) => <Button {...args} />;
```
+добавить testFn для onClick

Robocotik replied:
выглядит довольно нагромажденно, предлагаю раздел Story из доки итса к прочтению

coma8765 commented on src/components/Button/button.tsx:
```diff
@@ -0,0 +1,21 @@
+const Button: React.FC<ButtonProps> = ({ label, disabled = false, color = 'primary' }) => {
+    const baseStyles = 'px-4 py-2 font-headline-small rounded';
+    const filledStyles = 'bg-color-red text-white w-full max-w-lg h-14 p-4 rounded-2xl gap-4';
+    const borderedStyles = 'font-title-large text-white border-color-grey rounded-2xl w-44 h-16 p-4 gap-16 border-2';
```
гуд, но вынеси это в функцию getClassNames

coma8765 replied:
и желательно всё со стилями в `Button.classes.ts`

coma8765 replied:
@Robocotik добавь этот пример в доку

Robocotik commented on src/components/Button/button.tsx:
```diff
@@ -0,0 +1,21 @@
+import '../Text/index'
+const Button: React.FC<ButtonProps> = ({ label, disabled = false, color = 'primary' }) => {
+    const baseStyles = 'px-4 py-2 font-headline-small rounded';
+    const filledStyles = 'bg-color-red text-white w-full max-w-lg h-14 p-4 rounded-2xl gap-4';
```
"bg-color-red" твой кастомный цвет и так лежит в tailwind в разделе цвета, ты не сможешь ее заюзать еще где-то , лучше назвать перменную либо как фигме, либо по тону цвета (red-550), либо по смыслу (red-light)

coma8765 commented on src/components/Button/button.tsx:
```diff
@@ -0,0 +1,21 @@
+        <button
+            className={`${baseStyles} ${buttonStyles} ${disabled}`}
+            disabled={disabled}
+        >
```
прокинуть остальные пропсы

coma8765 commented on src/components/Button/button.props.ts:
```diff
@@ -0,0 +1,5 @@
+export interface ButtonProps {
+    label:string;
+    disabled?:boolean;
+    color?: 'filled' | 'bordered';
```
почему color? если это вариант

coma8765 commented on src/stories/button.css:
откатить, мы удалим папку `stories` скоро

sgrisshk committed:
fix (elements): fix some mistakes mentioned in review

add testFn for onClick and fix Story format (button.stories.tsx),changed color to variant and added special extension (button.props.ts), add button.classes.ts for styles and getClassname function, {...props} (button.tsx)

coma8765 reviewed (changes requested):
мне оч нравится, только один момент, название файла компонента с большой буквы, в том числе Button.props.ts, Button.stories.tsx и тп

sgrisshk committed:
rename (files): fix filename style

rename filenames styles of Button component

Robocotik reviewed (approved):
В целом ничего сильного нового от моего последнего ревью не появилось, остальное вроде исправлено, как я понимаю. Мне нравится в целом тоже

coma8765 reviewed (approved):
Гуд
